### PR TITLE
[skrifa] autohint bug fixes to match FT

### PR DESCRIPTION
**Pull request by dfrg**
This is a largish collection of small changes that bring the autohinter into a state where we match with FreeType for at least the Google Fonts and Noto collections.

Also updates fauntlet to use the new hinting API.

Builds on #1144 which should land first.

### CODE REVIEW

**Comment by dfrg** [comment]
cc @drott

**Review by drott — APPROVED** [review]
I can see the hours that went into finding those differences, thanks for bringing this in alignment with FT. LGTM.

File: skrifa/src/outline/autohint/hint.rs [hunk 1]
```diff
@@ -117,23 +117,45 @@ pub(crate) fn align_strong_points(outline: &mut Outline, axis: &mut Axis) -> Opt
...
-                Ordering::Less => max_ix = mid_ix,
-                Ordering::Greater => min_ix = mid_ix + 1,
-                Ordering::Equal => {
-                    // We are on an edge
+        // Find enclosing edges; for a small number of edges, use a linear
+        // search.
+        // Note: this is actually critical for matching FreeType in cases where
+        // we have more than one edge with the same fpos. When this happens,
+        // linear and binary searches can produce different results.
+        // See <https://gitlab.freedesktop.org/freetype/freetype/-/blob/57617782464411201ce7bbc93b086c1b4d7d84a5/src/autofit/afhints.c#L1489>
```
[the remark this answers is not in the document]
Comment (dfrg) [reply skrifa/src/outline/autohint/hint.rs:125]: Took me a while to find this one :)

File: skrifa/src/outline/autohint/latin/hint.rs [hunk 2]
```diff
@@ -576,25 +585,26 @@ fn stem_width(
...
-                        dist += 54;
-                    } else {
-                        dist += delta;
-                    }
+            }
+        }
+        if group != ScriptGroup::Default {
+            // Divergent CJK behavior
+            // See <https://gitlab.freedesktop.org/freetype/freetype/-/blob/57617782464411201ce7bbc93b086c1b4d7d84a5/src/autofit/afcjk.c#L1544>
+            if dist < 54 {
```
[the remark this answers is not in the document]
Comment (dfrg) [reply skrifa/src/outline/autohint/latin/hint.rs:593]: I’ll do the best I can. Updated #1141 to include this.

File: fauntlet/src/compare_glyphs.rs [hunk 3]
```diff
@@ -13,6 +15,20 @@ pub fn compare_glyphs(
         // Don't run on bitmap fonts (yet)
         return true;
     }
+    if let Some(Hinting::Auto(_)) = options.hinting {
+        // This font is a pathological case for autohinting due to the
```
[the remark this answers is not in the document]
Comment (dfrg) [reply fauntlet/src/compare_glyphs.rs:19]: Sgtm.

Filed #1147 and added a link in the comment

**Pull request merged**